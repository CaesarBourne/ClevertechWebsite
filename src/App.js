import React from 'react';
import { AppProviders } from './AppProviders';

import { Layout } from './components/Layout/layout';

export const App = () => (
  <AppProviders>
    <Layout />
  </AppProviders>
);
