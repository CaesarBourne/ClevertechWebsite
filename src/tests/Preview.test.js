import React from 'react';
import { fireEvent, render, getByText as staticGetByText } from '@testing-library/react'

import { App } from '../App';

describe('Preview', () => {
  it('should display empty preview if no components were added', () => {
    // given
    const { getByTestId } = render(<App />)

    // then
    const preview = getByTestId('components-preview')
    expect(preview).toMatchSnapshot()
  })

  it('should display a link component in the preview after it was added', () => {
    // given
    const { getByText, getByLabelText, getByTestId } = render(<App />)

    // when
    fireEvent.click(getByText('link'))
    fireEvent.change(getByLabelText('label'), { target: { value: "The Site" } })
    fireEvent.change(getByLabelText('href'), { target: { value: "www.site.com" } })
    fireEvent.click(getByText('Apply'))

    // then
    const preview = getByTestId('components-preview')
    expect(preview).toMatchSnapshot()
  })

  it('should display again an empty preview after a link was added and later deleted', () => {
    // given
    const { getByText, getByLabelText, getByTestId } = render(<App />)

    // when
      // add element
    fireEvent.click(getByText('link'))
    fireEvent.change(getByLabelText('label'), { target: { value: "The Site" } })
    fireEvent.change(getByLabelText('href'), { target: { value: "www.site.com" } })
    fireEvent.click(getByText('Apply'))
      // remove element
    const newElement = getByText('The Site').parentElement
    fireEvent.click(staticGetByText(newElement, 'click to edit'))
    fireEvent.click(getByText('Remove'))

    // then
    const preview = getByTestId('components-preview')
    expect(preview).toMatchSnapshot()
  })
});
