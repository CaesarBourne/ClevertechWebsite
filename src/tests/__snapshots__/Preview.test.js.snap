// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview should display a link component in the preview after it was added 1`] = `
<div
  class="components-preview"
  data-testid="components-preview"
>
  <div
    class="components-wrapper"
  >
    <a
      href="www.site.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      The Site
    </a>
    <span
      class="edit"
    >
      click to edit
    </span>
  </div>
</div>
`;

exports[`Preview should display again an empty preview after a link was added and later deleted 1`] = `
<div
  class="components-preview"
  data-testid="components-preview"
/>
`;

exports[`Preview should display empty preview if no components were added 1`] = `
<div
  class="components-preview"
  data-testid="components-preview"
/>
`;
